docling_core/types/doc/page.py

```diff
@@ -0,0 +1,1292 @@
+"""Datastructures for PaginatedDocument."""
+
+import json
+import logging
+import math
+import re
+from enum import Enum
+from pathlib import Path
+from typing import Annotated, Dict, Iterator, List, NamedTuple, Optional, Tuple, Union
+
+import numpy as np
+from PIL import Image as PILImage
+from PIL import ImageColor, ImageDraw, ImageFont
+from PIL.ImageFont import FreeTypeFont
+from pydantic import AnyUrl, BaseModel, Field, model_validator
+
+from docling_core.types.doc.base import BoundingBox, CoordOrigin
+from docling_core.types.doc.document import ImageRef
+
+# Configure logging
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s"
+)
+
+
+class TextCellUnit(str, Enum):
+    """Enumeration of text cell units for segmented PDF page processing."""
+
+    CHAR = "char"
+    WORD = "word"
+    LINE = "line"
+
+    def __str__(self) -> str:
+        """Return string representation of the enum value."""
+        return str(self.value)
+
+
+class PdfPageBoundaryType(str, Enum):
+    """Enumeration of PDF page boundary types."""
+
+    ART_BOX = "art_box"
+    BLEED_BOX = "bleed_box"
+    CROP_BOX = "crop_box"
+    MEDIA_BOX = "media_box"
+    TRIM_BOX = "trim_box"
+
+    def __str__(self) -> str:
+        """Return string representation of the enum value."""
+        return str(self.value)
+
+
+ColorChannelValue = Annotated[int, Field(ge=0, le=255)]
+
+
+class ColorRGBA(BaseModel):
+    """Model representing an RGBA color value."""
+
+    r: ColorChannelValue
+    g: ColorChannelValue
+    b: ColorChannelValue
+    a: ColorChannelValue = 255
+
+    def as_tuple(self) -> tuple[int, int, int, int]:
+        """Return the color components as a tuple."""
+        return (self.r, self.g, self.b, self.a)
+
+    def __iter__(self):
+        """Yield the color components for iteration."""
+        yield from (self.r, self.g, self.b, self.a)
+
+
+class Coord2D(NamedTuple):
```

Comment:
Where do we use this?

Reply:
Now we do. Since it is a named tuple it will serialize as a regular tuple, not as an object.